src/components/SettingsTreasuriesBlockItemButton.vue

```diff
@@ -0,0 +1,65 @@
+<script setup lang="ts">
+import { TreasuryWallet } from '@/helpers/interfaces';
+import { Network } from '@/plugins/oSnap/types';
+import { getIsOsnapEnabled } from '@/plugins/oSnap/utils/getters';
+
+const props = defineProps<{
+  treasury: TreasuryWallet;
+  treasuryIndex: number;
+  isViewOnly?: boolean;
+  hasOsnapPlugin: boolean;
+}>();
+
+const emit = defineEmits<{
+  removeTreasury: [index: number];
+  editTreasury: [index: number];
+  configureOsnap: [index: number, isEnabled: boolean];
+}>();
+
+const isOsnapEnabled = ref(false);
+
+async function updateIsOsnapEnabled() {
```

Comment:
data fetching does not run if the osnap plugin is not installed, so other users will not experience increased traffic